text/src/autogluon/text/text_prediction/mx/models.py

```diff
@@ -1125,6 +1204,35 @@ def train(self, train_data, tuning_data,
         self._net = net
         mx.npx.waitall()
 
+        # Clean cache
+        try:
+            os.remove(os.path.join(cache_path, 'cache_train_dataframe.pd.pkl'))
+            os.remove(os.path.join(cache_path, 'cache_tuning_dataframe.pd.pkl'))
+            if continue_training:
+                os.remove(os.path.join(cache_path, 'old_net.params'))
+                os.remove(os.path.join(cache_path, 'preprocessor.pkl'))
+            os.rmdir(cache_path)
+        except OSError as e:
+            logger.info(f'Failed to remove the cache directory at "{cache_path}"')
+
+        # Clean up the trained model weights since the model weights are loaded in the network.
+        clean_up_param_l = []
+        for best_id in range(cfg.optimization.nbest):
+            nbest_path = os.path.join(best_model_saved_dir_path, f'nbest_model{best_id}.params')
+            if os.path.exists(nbest_path):
+                clean_up_param_l.append(nbest_path)
+        best_model_weight = os.path.join(best_model_saved_dir_path, 'best_model.params')
+        if os.path.exists(best_model_weight):
+            clean_up_param_l.append(best_model_weight)
+        if cfg.model.use_avg_nbest:
+            clean_up_param_l.append(avg_nbest_path)
+
+        for ele in clean_up_param_l:
+            try:
+                os.remove(ele)
+            except OSError as e:
+                logger.info(f'Failed to remove the model weight from "{ele}".')
+
```

Comment:
I'm afraid of bugs sneaking by in future changes because of the complexity of this cleaning. Is there any way for the model to keep track of all the cached files in an object so that instead of hard-coding all of these filenames it is instead fetched from an object? I'm ok with this being added as a TODO but it is something we should be mindful of, this logic makes it harder to work on this model code without adding bugs.

Reply:
That's a valid concern. The appropriate approach might to create a temporary directory with unique id that stores everything inside and will be cleaned afterwards. I will address it in the later PR that adds pytorch.

Reply:
@Innixma I should have fixed the logic with uuid. In fact, we can later consider to create temporary space in S3. This will be useful for large datasets which cannot be stored in disk.